autodist/simulator/rankrnn_simulator.py

```diff
@@ -0,0 +1,1027 @@
+from arion.simulator.utils import get_dense_var_bits, get_sparse_var_bits, GIGABITS
+from arion.simulator.utils import _resolve_device_address, _max_num_local_replica, _num_local_replica, _resolved_devices_on_diff_machine
+from arion.strategy.random_sample_strategy import VariableHelper, PartHelper
+from arion.strategy.base import Strategy
+from arion.resource_spec import ResourceSpec
+from arion.cluster import SSHCluster
+from arion.kernel.device.resolver import DeviceResolver
+from arion.kernel.partitioner import PartitionerConfig
+from arion.simulator.models.predefined_simulator import PredefinedSimulator
+
+import torch
+import torch.nn as nn
+
+import multiprocessing
+from multiprocessing import Process, Queue
+
+TORCH_DEVICE = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+# feature settings
+MAX_NUM_WORKERS = 16
+MAX_NUM_GROUPS = 600
+MAX_NUM_VARS = 500
+MAX_NUM_PARS = 1500
+FEATURE_SIZE = MAX_NUM_WORKERS+MAX_NUM_GROUPS+15
+
+# model size
+PARTITION_MLP_HIDDEN = 128
+PARTITION_MLP_OUT = 32
+STEM_RNN_HIDDEN = 128
+BIDIECTIONAL = True
+BATCH_SIZE = 96
+
+NUM_RNN_LAYERS = 3
+SCORE_TH = 0.005
+LR = 2e-3
+WD = 3e-4
+DATA_AUG = False
+IN_LAYERS = 2
+OUT_LAYERS = 1
+
+# ncf used:
+# ~/projects/pycharm/zhijie/5-9-2020/oceanus-zhijie/arion/simulator/models/model_train_on_ncf-orca_new.ckpt 0.9020
+# noaug
+# PARTITION_MLP_HIDDEN = 128
+# PARTITION_MLP_OUT = 32
+# STEM_RNN_HIDDEN = 128
+# BIDIECTIONAL = True
+# NUM_RNN_LAYERS = 4
+# BATCH_SIZE = 64
+# LR = 1e-3
+# WD = 4e-4
+
+# vgg used:
+# ~/projects/pycharm/zhijie/5-9-2020/oceanus-zhijie/arion/simulator/models/model_train_on_vgg16-orca_new_new_new.ckpt 0.8374
+# noaug
+# PARTITION_MLP_HIDDEN = 128
+# PARTITION_MLP_OUT = 32
+# STEM_RNN_HIDDEN = 128
+# BIDIECTIONAL = True
+# NUM_RNN_LAYERS = 3
+# BATCH_SIZE = 64
+# LR = 1e-3
+# WD = 3e-4
+
+GRAPH_ITEM_PATHS = {'ncf':'/users/hzhang2/projects/pycharm/zhijie/5-5-2020/original_graph_item',
+                'densenet121': '/users/hzhang2/projects/pycharm/zhijie/graph_items/densenet121_original_graph_item',
+                'inceptionv3': '/users/hzhang2/projects/pycharm/zhijie/graph_items/inceptionv3_original_graph_item',
+                'resnet101': '/users/hzhang2/projects/pycharm/zhijie/graph_items/resnet101_original_graph_item',
+                'resnet50': '/users/hzhang2/projects/pycharm/zhijie/graph_items/resnet50_original_graph_item',
+                'vgg16': '/users/hzhang2/projects/pycharm/zhijie/graph_items/vgg16_original_graph_item',
+                'bert_12l': '/users/hzhang2/projects/pycharm/zhijie/graph_items/bert_original_graph_item_12l',
+                'bert_6l': '/users/hzhang2/projects/pycharm/zhijie/graph_items/bert_original_graph_item_6l',
+                'bert_3l': '/users/hzhang2/projects/pycharm/zhijie/graph_items/bert_original_graph_item_3l',
+                'bert_large': '/users/hzhang2/projects/pycharm/zhijie/graph_items/bert_original_graph_item_large'}
```

Comment:
these are all specific to your computer right now

Reply:
yeah WIP, will have more updates down the road.